clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3582,21 +3407,55 @@ static bool InvalidWorkGroupSizeAttrs(const Expr *MGValue, const Expr *XDim,
            ZDimExpr->getResultAsAPSInt() != 1));
 }
 
-// If the [[intel::max_work_group_size(X, Y, Z)]] attribute is specified on
-// a declaration along with [[sycl::reqd_work_group_size(X1, Y1, Z1)]]
-// attribute, check to see if values of reqd_work_group_size arguments are
+// If the 'reqd_work_group_size' attribute is specified on
+// a declaration along with 'max_work_group_size' attribute,
+// check to see if values of reqd_work_group_size arguments are
 // equal or less than values of max_work_group_size attribute arguments.
-static bool checkWorkGroupSizeAttrValues(const Expr *RWGS, const Expr *MWGS) {
+// If the 'reqd_work_group_size' attribute is specified on
+// a declaration along with 'reqd_work_group_size' attribute,
+// check to see if values of reqd_work_group_size arguments are
+// equal or greater than values of 'reqd_work_group_size' attribute arguments.
```

Comment:
I can't quite make out what this comment is supposed to say.  One of them should be 'max_work_group_size', I think.

Reply:
@premanandrao, we are checking two mutual diagnostics here:

1. // / If the 'reqd_work_group_size' attribute is specified on
// a declaration along with another 'reqd_work_group_size' attribute,
// check to see if values of reqd_work_group_size arguments are
// equal or greater than values of 'reqd_work_group_size' attribute arguments.
Example:
```
class Functor32 {
public:
  [[sycl::reqd_work_group_size(32, 1, 1)]] // expected-note {{conflicting attribute is here}}
  [[sycl::reqd_work_group_size(1, 1, 32)]] void
  // expected-note@+2{{conflicting attribute is here}}
  // expected-error@+1{{'reqd_work_group_size' attribute conflicts with 'reqd_work_group_size' attribute}}
  [[sycl::reqd_work_group_size(32, 1, 1)]] [[sycl::reqd_work_group_size(1, 1, 32)]] void
  operator()() const {}
};
```
2. // If the 'reqd_work_group_size' attribute is specified on
// a declaration along with 'max_work_group_size' attribute,
// check to see if values of reqd_work_group_size arguments are
// equal or less than values of max_work_group_size attribute arguments.

Example:
```
[[sycl::reqd_work_group_size(64, 64, 64)]] void f12(); // expected-note {{conflicting attribute is here}}
[[intel::max_work_group_size(16, 16, 16)]] void f12(); // expected-error {{'max_work_group_size' attribute conflicts with 'reqd_work_group_size' attribute}}

```

Reply:
Thanks for the example.   Let me see if I can help make the comment better.
Which is the error case:
```
[[sycl::reqd_work_group_size(1, 1, 1)]] [[sycl::reqd_work_group_size(2, 2, 2)]] void
  operator()() const {}
```
or
```
[[sycl::reqd_work_group_size(2, 2, 2)]] [[sycl::reqd_work_group_size(1, 1, 1)]] void
  operator()() const {}
```
?

Reply:
>>[[sycl::reqd_work_group_size(1, 1, 1)]] [[sycl::reqd_work_group_size(2, 2, 2)]] void
  operator()() const {}

Error case
clang -cc1 -fsycl-is-device b5.cpp
b5.cpp:2:16: error: non-member function cannot have 'const' qualifier
  operator()() const {}
               ^~~~~~
b5.cpp:1:43: error: 'reqd_work_group_size' attribute conflicts with 'reqd_work_group_size' attribute
[[sycl::reqd_work_group_size(2, 2, 2)]] [[sycl::reqd_work_group_size(1, 1, 1)]] void
                                          ^
b5.cpp:1:3: note: conflicting attribute is here
[[sycl::reqd_work_group_size(2, 2, 2)]] [[sycl::reqd_work_group_size(1, 1, 1)]] void
  ^
b5.cpp:2:3: error: overloaded 'operator()' must have at least one parameter of class or enumeration type
  operator()() const {}
  ^
3 errors generated.

cat b5.cpp
[[sycl::reqd_work_group_size(2, 2, 2)]] [[sycl::reqd_work_group_size(1, 1, 1)]] void
  operator()() const {}

clang -cc1 -fsycl-is-device b6.cpp
b6.cpp:2:16: error: non-member function cannot have 'const' qualifier
  operator()() const {}
               ^~~~~~
b6.cpp:1:43: warning: attribute 'reqd_work_group_size' is already applied with different arguments [-Wignored-attributes]
[[sycl::reqd_work_group_size(1, 1, 1)]] [[sycl::reqd_work_group_size(2, 2, 2)]] void
                                          ^
b6.cpp:1:3: note: previous attribute is here
[[sycl::reqd_work_group_size(1, 1, 1)]] [[sycl::reqd_work_group_size(2, 2, 2)]] void
  ^
b6.cpp:2:3: error: overloaded 'operator()' must have at least one parameter of class or enumeration type
  operator()() const {}
  ^
1 warning and 2 errors generated.
cat b6.cpp
[[sycl::reqd_work_group_size(1, 1, 1)]] [[sycl::reqd_work_group_size(2, 2, 2)]] void
  operator()() const {}

Reply:
I will double check this behavior to make sure to match with existing implementation.